hack/generator/pkg/astmodel/validated_type.go

```diff
@@ -156,6 +156,16 @@ func (v *ValidatedType) ElementType() Type {
 	return v.element
 }
 
+func (v *ValidatedType) WithElement(t Type) Type {
+	if v.element.Equals(t) {
+		return v
+	}
+
+	result := *v
+	result.element = t
+	return &result
+}
```

Comment:
When cloning types, hadn't we agreed on using a memberwise `copy()` func to ensure all the fields were handled correctly? ('cause we had some issues early on with "copies" that shared state.)

Reply:
Here we don’t touch any of the interior-mutable types like maps or slices, so it is safe. I’d prefer not to pessimize this…

Reply:
Fair enough.